TESTS/netsocket/udp/udpsocket_echotest.cpp

```diff
@@ -94,6 +94,9 @@ void UDPSOCKET_ECHOTEST_impl(bool use_sendto)
             } else {
                 sent = sock.send(tx_buffer, pkt_s);
             }
+            if (sent == NSAPI_ERROR_NO_MEMORY) {
```

Comment:
Don't we need to do the same for the non-blocking implementation?

Reply:
Lets re-think this..

Your `sleep_for()` might be fixing the issue, but only because it adds some delay.

But within that delay, there might be other packets coming (duplicates), cumulating our input buffers.

So instead of adding a error handling here, we should follow a generic UDP application priciples:
1. Ignore send errors.
2. Don't retry immediately.
2. Let upper layer or application logic handle the retries and timeouts.

Our application logic is a few lines down. On line 103 on original. There is `continue` call, which immediately retries. So to fix this, I propose that we remove the `continue` and allow it to flow directly into the `receive()` loop that starts from line 106.

Notice that in case the packet did not really go out, we get nothing back, and this loop will restart. So the retry logic is there already. But instead of immediately retrying, it would allow `recv()` or `recvfrom()` to drain potential duplicate packets away from input buffers.

So, we should be OK with only one "sleep" within this logic. If this does not work, make sure that `SOCKET_TIMEOUT` is long enough that we do sleep in `recvfrom()` and while doing that, drain our input buffers.

Reply:
Done as you suggested. Both continue and sleep removed.
It's also working fine.